README.md

```diff
@@ -219,18 +235,73 @@ void main() {
 }
 ```
 
-This can be run using:
+You can launch this app directly from VS Code, or by running:
 
 ```bash
 flutter run -t path/to/my_scenes.stager_app.g.dart
 ```
 
-You can launch to a specific scene by providing the name of the scene as an argument:
+If your Stager app consists of multiple Scenes, you can launch to a specific scene by providing the name of the scene as an argument:
 
 ```bash
 flutter run -t path/to/my_scenes.stager_app.g.dart --dart-define='Scene=No Posts'
 ```
 
+### Adding your own environment controls
+
+Stager's control panel comes with a generally useful set of controls that enable you to toggle dark mode, adjust text scale, etc. However, it is very likely that your app has unique environment properties that would be useful to adjust. To support this, Scenes have an overridable `environmentControlBuilders` property which allows you to add custom widgets to the default set of environment manipulation controls.
+
+A very simple example:
+
+```dart
+class CounterScene extends Scene {
+  // Define a custom property that is consumed by the Scene.
+  // This might be something as simple as an int, or something
+  // more complex, like a mocked network service.
+  int count = 0;
+
+  @override
+  Widget build() {
+    return EnvironmentAwareApp(
+      home: Scaffold(
+        body: Center(
+          child: Text(count.toString()),
+        ),
+      ),
+    );
+ }
+
+  @override
+  String get title => 'Counter';
+
+  @override
+  List<EnvironmentControlBuilder> get environmentControlBuilders => [
+        (BuildContext context, VoidCallback rebuildScene) {
+          // NumberStepperControl is a widget provided by Stager for convenience.
+          // However, your EnvironmentControlBuilder can return any arbitrary widget.
+          return NumberStepperControl(
+            title: const Text('Count'),
+            value: count,
+            onDecrementPressed: () {
+              count -= 1;
+
+              // Call rebuildScene to update the Scene with count's new value
+              rebuildScene();
+            },
+            onIncrementPressed: () {
+              count += 1;
+
+              // Call rebuildScene to update the Scene with count's new value
+              rebuildScene();
+            },
+          );
+        }
+      ];
+}
+```
+
+More complex examples can be found in `WithPostsScene` in `example/lib/pages/posts_list/posts_list_page_scenes.dart` and `PostDetailPageScene` in `example/lib/pages/post_detail/post_detail_page_scenes.dart`.
```

Comment:
I can imagine there will be scenarios where it doesn't make sense to call rebuildScene(), e.g. if you were to change the return value of a fake RPC, it might be necessary for the user to 'pull to refresh' to see the new data. Or if the scene shows a stateful widget that receives data from a subscription to a stream that is controlled by the environment controls. Maybe it'd make sense to mention that here?